freud/data.py

```diff
@@ -0,0 +1,163 @@
+# Copyright (c) 2010-2019 The Regents of the University of Michigan
+# This file is from the freud project, released under the BSD 3-Clause License.
+
+R"""
+The :class:`freud.data` module provides certain sample data sets and utility
+functions that are useful for testing and examples.
+"""
+
+import numpy as np
+import freud
+
+
+class UnitCell(object):
+    """Class to represent the unit cell of a crystal structure.
+
+    This class represents the unit cell of a crystal structure, which is
+    defined by a lattice and a basis. It provides the basic attributes of the
+    unit cell as well as enabling the generation of systems of points
+    (optionally with some noise) from the unit cell.
+
+    Args:
+        box:
+            A box-like object (see :meth:`~freud.box.Box.from_box`) containing
+            the lattice vectors of the unit cell.
+        basis_positions ((:math:`N_{points}`) :class:`numpy.ndarray`):
+            The basis of the unit cell in fractional coordinates
+            (Default value = ``[[0, 0, 0]]``.
+    """
+    def __init__(self, box, basis_positions=[[0, 0, 0]]):
+        self._box = freud.box.Box.from_box(box)
+        self._basis_positions = np.dot(basis_positions, self.box.to_matrix().T)
+
+    def to_system(self, num_replicas=1, scale=1, sigma_noise=0):
+        """Generate a system from the unit cell.
+
```

Comment:
Needs a bit of a description of how the scaling / replication / noise addition works.

Reply:
e.g.
```
The basis positions are transformed into absolute coordinates for every replica requested.
The box and positions are expanded by a factor of ``scale`` about the origin.
Finally, noise with standard deviation ``sigma_noise`` is added to the positions and the
positions are wrapped back into the box.
```